diff -ruN a/edma_drv/bma_include.h b/edma_drv/bma_include.h
--- a/edma_drv/bma_include.h	2026-08-19 10:06:48.000000000 +0000
+++ b/edma_drv/bma_include.h	2026-08-26 10:05:20.701954009 +0000
@@ -32,6 +32,11 @@
 #include <linux/version.h>
 #include <linux/semaphore.h>
 #include <linux/sched.h>
+#include <linux/timer.h>
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 16, 0)
+#define del_timer_sync(timer) timer_delete_sync(timer)
+#endif
 
 #define UNUSED(x) (x = x)
 #define KBOX_FALSE (-1)
@@ -93,7 +98,7 @@
 	} specific;
 };
 
-#if defined(timer_setup) && defined(from_timer)
+#if defined(timer_setup)
 #define HAVE_TIMER_SETUP
 #endif
 
diff -ruN a/edma_drv/edma_host.c b/edma_drv/edma_host.c
--- a/edma_drv/edma_host.c	2026-08-19 10:06:48.000000000 +0000
+++ b/edma_drv/edma_host.c	2026-08-26 10:05:20.703954018 +0000
@@ -882,7 +882,7 @@
 #ifdef HAVE_TIMER_SETUP
 static void edma_host_timeout(struct timer_list *t)
 {
-	struct edma_host_s *edma_host = from_timer(edma_host, t, timer);
+	struct edma_host_s *edma_host = timer_container_of(edma_host, t, timer);
 #else
 static void edma_host_timeout(unsigned long data)
 {
@@ -903,8 +903,8 @@
 #ifdef HAVE_TIMER_SETUP
 static void edma_host_heartbeat_timer(struct timer_list *t)
 {
-	struct edma_host_s *edma_host = from_timer(edma_host, t,
-						    heartbeat_timer);
+	struct edma_host_s *edma_host = timer_container_of(edma_host, t,
+							    heartbeat_timer);
 #else
 static void edma_host_heartbeat_timer(unsigned long data)
 {
@@ -953,7 +953,7 @@
 #ifdef HAVE_TIMER_SETUP
 static void edma_host_dma_timeout(struct timer_list *t)
 {
-	struct edma_host_s *edma_host = from_timer(edma_host, t, dma_timer);
+	struct edma_host_s *edma_host = timer_container_of(edma_host, t, dma_timer);
 #else
 static void edma_host_dma_timeout(unsigned long data)
 {
diff -ruN a/edma_drv/edma_host.h b/edma_drv/edma_host.h
--- a/edma_drv/edma_host.h	2026-08-19 10:06:48.000000000 +0000
+++ b/edma_drv/edma_host.h	2026-08-26 10:05:20.705954027 +0000
@@ -35,73 +35,6 @@
 #define UNUSED
 #endif
 
-/* vm_flags in vm_area_struct, see mm_types.h. */
-#define VM_NONE		0x00000000
-
-#define VM_READ		0x00000001	/* currently active flags */
-#define VM_WRITE	0x00000002
-#define VM_EXEC		0x00000004
-#define VM_SHARED	0x00000008
-
-#define VM_MAYREAD	0x00000010	/* limits for mprotect() etc */
-#define VM_MAYWRITE	0x00000020
-#define VM_MAYEXEC	0x00000040
-#define VM_MAYSHARE	0x00000080
-
-#define VM_GROWSDOWN	0x00000100	/* general info on the segment */
-/* Page-ranges managed without "struct page", just pure PFN */
-#define VM_PFNMAP	0x00000400
-#define VM_DENYWRITE	0x00000800	/* ETXTBSY on write attempts.. */
-
-#define VM_LOCKED	0x00002000
-#define VM_IO           0x00004000	/* Memory mapped I/O or similar */
-
-					/* Used by sys_madvise() */
-#define VM_SEQ_READ	0x00008000	/* App will access data sequentially */
-/* App will not benefit from clustered reads */
-#define VM_RAND_READ	0x00010000
-
-#define VM_DONTCOPY	0x00020000	/* Do not copy this vma on fork */
-#define VM_DONTEXPAND	0x00040000	/* Cannot expand with mremap() */
-#define VM_ACCOUNT	0x00100000	/* Is a VM accounted object */
-#define VM_NORESERVE	0x00200000	/* should the VM suppress accounting */
-#define VM_HUGETLB	0x00400000	/* Huge TLB Page VM */
-#define VM_NONLINEAR	0x00800000	/* Is non-linear (remap_file_pages) */
-#define VM_ARCH_1	0x01000000	/* Architecture-specific flag */
-#define VM_DONTDUMP	0x04000000	/* Do not include in the core dump */
-/* Can contain "struct page" and pure PFN pages */
-#define VM_MIXEDMAP	0x10000000
-
-#define VM_MERGEABLE	BIT(31)		/* KSM may merge identical pages */
-
-#if defined(CONFIG_X86)
-/* PAT reserves whole VMA at once (x86) */
-#define VM_PAT		VM_ARCH_1
-#elif defined(CONFIG_PPC)
-#define VM_SAO		VM_ARCH_1	/* Strong Access Ordering (powerpc) */
-#elif defined(CONFIG_PARISC)
-#define VM_GROWSUP	VM_ARCH_1
-#elif defined(CONFIG_METAG)
-#define VM_GROWSUP	VM_ARCH_1
-#elif defined(CONFIG_IA64)
-#define VM_GROWSUP	VM_ARCH_1
-#elif !defined(CONFIG_MMU)
-#define VM_MAPPED_COPY	VM_ARCH_1 /* T if mapped copy of data (nommu mmap) */
-#endif
-
-#ifndef VM_GROWSUP
-#define VM_GROWSUP	VM_NONE
-#endif
-
-#ifndef VM_STACK_DEFAULT_FLAGS	/* arch can override this */
-#define VM_STACK_DEFAULT_FLAGS VM_DATA_DEFAULT_FLAGS
-#endif
-
-#define VM_READHINTMASK			(VM_SEQ_READ | VM_RAND_READ)
-#define VM_NORMAL_READ_HINT(v)		(!((v)->vm_flags & VM_READHINTMASK))
-#define VM_SEQUENTIAL_READ_HINT(v)	((v)->vm_flags & VM_SEQ_READ)
-#define VM_RANDOM_READ_HINT(v)		((v)->vm_flags & VM_RAND_READ)
-
 #define REG_PCIE1_DMAREAD_ENABLE	0xa18
 #define SHIFT_PCIE1_DMAREAD_ENABLE	0
 
diff -ruN a/veth_drv/veth_hb.c b/veth_drv/veth_hb.c
--- a/veth_drv/veth_hb.c	2026-08-19 10:06:48.000000000 +0000
+++ b/veth_drv/veth_hb.c	2026-08-26 10:05:20.707954037 +0000
@@ -1144,8 +1144,8 @@
 static void veth_ethtool_get_drvinfo(struct net_device *dev,
 				     struct ethtool_drvinfo *info)
 {
-	strlcpy(info->driver, MODULE_NAME, sizeof(info->driver));
-	strlcpy(info->version, VETH_VERSION, sizeof(info->version));
+	strscpy(info->driver, MODULE_NAME, sizeof(info->driver));
+	strscpy(info->version, VETH_VERSION, sizeof(info->version));
 
 	info->n_stats = VETH_GLOBAL_STATS_LEN;
 }
