diff -ruN a/hibmc_drm_dp.c b/hibmc_drm_dp.c
--- a/hibmc_drm_dp.c	2026-08-19 10:06:47.000000000 +0000
+++ b/hibmc_drm_dp.c	2026-08-26 10:05:20.700954004 +0000
@@ -39,7 +39,7 @@
 }
 
 static int hibmc_dp_mode_valid(struct drm_connector *connector,
-			       struct drm_display_mode *mode,
+				       const struct drm_display_mode *mode,
 			       struct drm_modeset_acquire_ctx *ctx,
 			       enum drm_mode_status *status)
 {
diff -ruN a/hibmc_drm_drv.c b/hibmc_drm_drv.c
--- a/hibmc_drm_drv.c	2026-08-19 10:06:47.000000000 +0000
+++ b/hibmc_drm_drv.c	2026-08-26 10:05:20.698953995 +0000
@@ -13,11 +13,12 @@
 
 #include <linux/module.h>
 #include <linux/pci.h>
+#include <linux/aperture.h>
 
-#include <drm/drm_aperture.h>
+#include <drm/clients/drm_client_setup.h>
 #include <drm/drm_atomic_helper.h>
 #include <drm/drm_drv.h>
-#include <drm/drm_fbdev_generic.h>
+#include <drm/drm_fbdev_ttm.h>
 #include <drm/drm_gem_framebuffer_helper.h>
 #include <drm/drm_gem_vram_helper.h>
 #include <drm/drm_managed.h>
@@ -76,13 +77,13 @@
 	.driver_features	= DRIVER_GEM | DRIVER_MODESET | DRIVER_ATOMIC,
 	.fops			= &hibmc_fops,
 	.name			= "hibmc",
-	.date			= "20160828",
 	.desc			= "hibmc drm driver",
 	.major			= 1,
 	.minor			= 0,
 	.debugfs_init		= drm_vram_mm_debugfs_init,
 	.dumb_create            = hibmc_dumb_create,
 	.dumb_map_offset        = drm_gem_ttm_dumb_map_offset,
+	DRM_FBDEV_TTM_DRIVER_OPS,
 };
 
 static int __maybe_unused hibmc_pm_suspend(struct device *dev)
@@ -391,7 +392,7 @@
 	struct drm_device *dev;
 	int ret;
 
-	ret = drm_aperture_remove_conflicting_pci_framebuffers(pdev, &hibmc_driver);
+	ret = aperture_remove_conflicting_pci_devices(pdev, hibmc_driver.name);
 	if (ret)
 		return ret;
 
@@ -426,7 +427,7 @@
 		goto err_unload;
 	}
 
-	drm_fbdev_generic_setup(dev, 32);
+	drm_client_setup(dev, NULL);
 
 	return 0;
 
